waspc/data/Generator/templates/react-app/src/api.ts

```diff
@@ -1,44 +1,44 @@
-import axios from 'axios'
+import axios, { type AxiosError } from 'axios'
 import config from './config'
+import { storage } from './storage'
 
 const api = axios.create({
   baseURL: config.apiUrl,
 })
 
-const WASP_APP_AUTH_TOKEN_NAME = "authToken"
+const WASP_APP_AUTH_TOKEN_NAME = 'authToken'
 
-let authToken = null
-if (window.localStorage) {
-  authToken = window.localStorage.getItem(WASP_APP_AUTH_TOKEN_NAME)
-}
+let authToken: string | undefined = storage.get(WASP_APP_AUTH_TOKEN_NAME) as string | undefined
 
-export const setAuthToken = (token) => {
+export const setAuthToken = (token: unknown): void => {
   if (typeof token !== 'string') {
-    throw Error(`Token must be a string, but it was: {${typeof token}} ${token}.`)
+    throw Error(
+      `Token must be a string, but it was: {${typeof token}} ${token}.`
+    )
   }
   authToken = token
-  window.localStorage && window.localStorage.setItem(WASP_APP_AUTH_TOKEN_NAME, token)
+  storage.set(WASP_APP_AUTH_TOKEN_NAME, token)
 }
 
-export const clearAuthToken = () => {
+export const clearAuthToken = (): void => {
```

Comment:
**Nit:** If you feel like it, you can change these guys to regular function (reasoning: #487). If not, just ignore the comment.

Reply:
Good suggestion, I didn't do it since I didn't want to touch too many lines, but while we are at it, let's do it properly.